server/repository/src/db_diesel/temperature_breach.rs

```diff
@@ -87,37 +98,68 @@ impl<'a> TemperatureBreachRepository<'a> {
         let result = query
             .offset(pagination.offset as i64)
             .limit(pagination.limit as i64)
-            .load::<TemperatureBreachRow>(&self.connection.connection)?;
+            .load::<TemperatureBreachJoin>(&self.connection.connection)?;
 
         Ok(result.into_iter().map(to_domain).collect())
     }
-}
 
-type BoxedLogQuery = temperature_breach::BoxedQuery<'static, DBType>;
-
-fn create_filtered_query(filter: Option<TemperatureBreachFilter>) -> BoxedLogQuery {
-    let mut query = temperature_breach::table.into_boxed();
-
-    if let Some(filter) = filter {
-        apply_equal_filter!(query, filter.id, temperature_breach_dsl::id);
-        apply_equal_filter!(query, filter.sensor_id, temperature_breach_dsl::sensor_id);
-        apply_equal_filter!(query, filter.store_id, temperature_breach_dsl::store_id);
-        //apply_equal_filter!(query, filter.r#type, temperature_breach_dsl::type_);
-        apply_date_time_filter!(
-            query,
-            filter.start_timestamp,
-            temperature_breach_dsl::start_timestamp
-        );
-        apply_date_time_filter!(
-            query,
-            filter.end_timestamp,
-            temperature_breach_dsl::end_timestamp
-        );
-    }
-
-    query
+    pub fn create_filtered_query(
+        filter: Option<TemperatureBreachFilter>,
+    ) -> Result<BoxedTemperatureBreachQuery, RepositoryError> {
+        let mut query = temperature_breach_dsl::temperature_breach
+            .inner_join(sensor_dsl::sensor)
+            .left_join(location_dsl::location)
+            .into_boxed();
```

Comment:
I don't think you need the join here if you only want to do the filtering, i.e. it should work without the joins.

Reply:
I need to to load sensor and location in TemperatureBreachNode, I don't think it works without the joins...? 👀

Reply:
You are already using loaders in the TemperatureBreachNode, so these rows get joined here but aren't used anywhere...

Think both approaches would work, I guess in our current use case we always need sensor and location? However, not joining would make it more efficient if you don't need sensor and location. I would say having loaders for sensors and location should also be fast enough even if we always need sensor and location. Moreover, not joining would keep the repo more flexible (no unnecessary joins when data is not needed, e.g. in service validations (?)). So I would do it without joins (would also keep the repo code simpler).